_posts/2023-12-10-Getting Started with Rust SDK.md

```diff
@@ -0,0 +1,326 @@
+
+# Prerequisites
+### Install Rust and Cargo 
+The easiest way to get Cargo is to install the current stable release of Rust by using rustup. Installing Rust using rustup will also install cargo.
+On Linux and macOS systems, this is done as follows:
+
+> `curl https://sh.rustup.rs -sSf | sh`
+
+It will download a script, and start the installation. If everything goes well, you’ll see this appear:
+
+> `Rust is installed now. Great!`
+
+On Windows, download and run `rustup-init.exe`. It will start the installation in a console and present the above message on success.
+
+After this, you can use the rustup command to also install beta or nightly channels for Rust and Cargo.
+
+You can find more informatin on rust on [The Cargo Book](https://doc.rust-lang.org/cargo/getting-started/installation.html).
+
+### Starting a new Rust Project and installing the SDK
+You can start a new rust project through cargo with the following command: 
+> `cargo new resilientDB_test`
+
+You should now see a new folder generated named `resilientDB_test`.
+```
+$ cd resilientDB_test
+$ tree .
+.
+├── Cargo.toml
+└── src
+    └── main.rs
+
+1 directory, 2 files
+``` 
+You can get started with the Rust SDK in one of two ways: 
+1. You can simply run the following command:
+   ```
+   cargo add resilientdb_rust_sdk
+   ```
+2. You can also add the following line to your `Cargo.toml`
+   ```
+   resilientdb_rust_sdk = "0.1.1"
+   ```
+
+It's that easy! You are now good to go with our SDK. Make sure you have a ResilientDB instance setup, for more info on that follow this [blog](https://blog.resilientdb.com/2023/11/22/Deploying-ResilientDB.html). This tutorial also makes a graphql query to resdb, if you don't have graphql setup take a look at this other [blog]().
+
+### Dependencies
+
+You may need to install the following dependencies as our library has `future` return types, meaning some returns from our APIs would need to be unwrapped within an `async` function. 
+
+This is by design as we encourage users to handle errors on their end as many of the calls are made asynchrounously. Following good error handling practices would be beneficial to developing a secure application with ResilientDB. You can run obtain the dependencies through: 
+
+1. Running the following command
+   > `cargo add tokio serde serde_json reqwest`
+   
+2. Adding this to your Cargo.toml file:
+   ```rust
+    tokio = { version = "1", features = ["full"] }
+    serde = { version = "1.0", features = ["derive"] }
+    serde_json = "1.0"
+    reqwest = {version = "0.11.22", features = ["blocking", "json"]}
+   ```
+
+# Developement
+### Import the SDK
+
+Let's build a sample program that makes a transactions and then fetches transactions to and from your ResilientDB instance. In your `main.rs` file import the SDK like so: 
+```rust
+use resilientdb_rust_sdk::ResDB;
```

Comment:
I added this dependency in the file 
```
use resilientdb_rust_sdk::ResDB
```
and run
```
rustc src/main.rs
```
However, I got this error:
```
2 | use resilientdb_rust_sdk::ResDB;
  |     ^^^^^^^^^^^^^^^^^^^^ maybe a missing crate `resilientdb_rust_sdk`?
 ```
 
 Is there anything missing here?

Reply:
You are trying to use `rustc` which is the rust compiler. The rust compiler will need flags in order to properly import the right dependencies (Just like C++), which is why you are getting that error. I would recommend you use `cargo run src/main.rs` or stepping into `./src` and running `cargo run` this automatically passes the required flags to `rustc`. Let me know if that works.

Reply:
Looks good.
So cargo needs to be installed.